xpad.c

```diff
@@ -432,6 +441,12 @@ static const signed short xpad_btn_paddles[] = {
 	-1						/* terminating entry */
 };
 
+/* used for GHL dpad mapping */
+static const struct {int x; int y; } dpad_mapping[] = {
```

Comment:
can you use the same logic as here:
https://github.com/paroj/xpad/blob/2c087f90bb85799df02400f6f10ccdca70e02c6e/xpad.c#L1021-L1024

instead of this mapping table?